src/Library.php

```diff
@@ -216,7 +216,7 @@ public static function delete_expired_draft_orders() {
 	 * @param PaymentContext $context Holds context for the payment.
 	 * @param PaymentResult  $result  Result of the payment.
 	 */
-	public static function process_legacy_payment( PaymentContext $context, PaymentResult &$result ) {
+	public static function process_legacy_payment( PaymentContext $context, PaymentResult $result ) {
```

Comment:
Just checking; did you test that legacy payment methods could still append things to $result and that was seen by the caller?

Reply:
woops another artifact from initial iterations inthis pull, I put back.